### DISCUSSION TITLE
Set Java Version only for signal-cli

### DISCUSSION THREAD

**Discussion by xtourtemyx** [AsamK/signal-cli#1227] [Q&A] [answered]
Hello, does someone know any methods how i can set java 17 only for signal? Right now i have a unifi controller, which require java 11 and signal-cli, which require java 17

**Comment by bublath** [comment, marked as answer]
Change your JAVA_HOME environment variable to the base directory of your Java 17 installation (export JAVA_HOME=/.... ) before starting signal-cli
When starting as a daemon you need to add
Environment="JAVA_HOME=/..."
in your service file.
